File: contenidors/piles/recursivitat1.cc
#include <iostream>
#include <stack>
using namespace std;

void work(int n)
{
    stack<int> s;
    s.push(n);
    while (not s.empty()) {
        if (s.top() != 1) {
            int x = s.top();
            s.pop();
            cout << ' ' << x;
            s.push(x - 1);
            s.push(x - 1);
        } else {
            cout << ' ' << s.top();
            s.pop();
        }
    }
}

int main()
{
    int n;
    while (cin >> n) {
        work(n);
        cout << endl;
    }
}

File: contenidors/piles/recursivitat1.py
from easyinput import read
from collections import deque

stack = deque()

def work(n):
    stack.append(n)
    while not stack.empty():
        if stack.top() != 1:
            stack.pop()
            stack.append(stack.top() - 1)
            stack.append(stack.top() - 1)
        else:
            stack.pop()


def main():
    n = read(int)
    while n is not None:
        work(n)
        print()
        n = read(int)

main()
